### DISCUSSION TITLE
Caution: Fake MAX3232 DB9 RS232 TTL converter

### DISCUSSION THREAD

**Discussion by wavoigt** [GyroGearl00se/ESPHome-Froeling-Lambdatronic_3200#9] [General]
This Converters may have Chinese Fake MAX3232 Chips, with following consequences:
Internal crosstalk on the chip leads to a distorted "echo" on the tx line which may disturb the communication with Froeling 3200!!!
see also:
[https://blog.bogpeople.com/2015/01/fake-max3232-rs232-line-drivers.html](https://blog.bogpeople.com/2015/01/fake-max3232-rs232-line-drivers.html)

I changed the MAX3232 by the original Maxim part and all worked fine.
![MAX3232 Fake - Origin](https://github.com/GyroGearl00se/ESPHome-Froeling-Lambdatronic_3200/assets/24320737/b3ee0266-19e1-4a56-9925-6826ec5b23d1)

**Comment by GyroGearl00se** [comment]
Do you know any board with the genuine part where you can provide a link ?

Or do you suggest desoldering the IC from the current board and replacing it with the genuine ?


So far it seems like I haven't had any issues with my current module.

**Comment by wavoigt** [comment]
Sorry, all the interface boards i found might have the fake chip.
If you have problems with the communication with any device I suggest replacing the chip.
https://www.reichelt.de/rs232-2-treiber-2-empfaenger-so-16-max-3232-cse-p57869.html?&trstct=pos_0&nbc=1

**Comment by GyroGearl00se** [comment]
> Sorry, all the interface boards i found might have the fake chip.
> If you have problems with the communication with any device I suggest replacing the chip.
> https://www.reichelt.de/rs232-2-treiber-2-empfaenger-so-16-max-3232-cse-p57869.html?&trstct=pos_0&nbc=1

Ok. Thank you for the hint.
I'll keep that in mind 😃

Close this 'issue'?

**Comment by wavoigt** [comment]
Thx for the ESPHome Integration of Froeling 3200 :-)
This is exactly what i've been looking for.
I'd not close the issue, so that others are aware of those chinese fake chips.

**Comment by StyxyDog** [comment]
> This Converters may have Chinese Fake MAX3232 Chips, with following consequences: Internal crosstalk on the chip leads to a distorted "echo" on the tx line which may disturb the communication with Froeling 3200!!! see also: https://blog.bogpeople.com/2015/01/fake-max3232-rs232-line-drivers.html
> 
> I changed the MAX3232 by the original Maxim part and all worked fine.

Hello, I'm very interested in this project. I have a Froling PE1. I am based in Scotland and I am very nervous of connecting anything to the boiler that may cause an issue.  It is very expensive and slow to get Froling parts here from Austria! So maybe I should wait for summer before trying to get involved. 

When you say: 
> distorted "echo" on the tx line which may disturb the communication

is this an issue that might effect the proper operation of the boiler - or even potentially cause damage or is it just an issue on the HA side?

Thank you very much to @wavoigt for raising this issue and of course to @GyroGearl00se for embarking on such a project!

**Comment by wavoigt** [comment]
Don't worry, this issue only might affect the communication, it won't effect the proper operation of the boiler.
It won't cause damage to the boiler.

**Comment by TehDuckys** [comment]
I am trying to get this project to work, but can't get any information to esp32, everything is 0 or unknown. 
I bought exact rs232 ttl adapter as it is on first link (bogpeople). Should I try with a different one?
Also there was no information on which cable to connect from com2 to adapter, so I bought normal db9, but later checking in lambdatronic documentation I read it needs to be null modem cable. I unplugged com1 cable that should be null modem as per documentation and plugged it in com2 and added my rs232 on it.
Can someone give me some direction?

**Reply by wavoigt** [reply]
You have to use a normal RS232 cable (female-male, 1:1 wired).
Did you try to swap rx / tx wires?
I use the adapter from the picture above and i think i had to connect rx to pin16 and tx to pin17
(see issue 6)
